pkg/kubelet/dockertools/manager.go

```diff
@@ -280,8 +280,15 @@ func (dm *DockerManager) inspectContainer(dockerID, containerName, tPath string)
 		result.status.State.Running = &api.ContainerStateRunning{
 			StartedAt: util.NewTime(inspectResult.State.StartedAt),
 		}
-		if containerName == PodInfraContainerName && inspectResult.NetworkSettings != nil {
-			result.ip = inspectResult.NetworkSettings.IPAddress
+		if dm.networkPlugin.Name() != network.DefaultPluginName && containerName == PodInfraContainerName {
+			netStatus, err := dm.networkPlugin.Status(pod.Namespace, pod.Name, kubeletTypes.DockerID(dockerID))
+			if err == nil {
+				result.ip = netStatus.IPv4.String()
```

Comment:
Can primary be an attribute in the Interface Info rather than repeating it in NetworkStatus ?
What about routes ?  The container network plugins are allowing the routes to be published. May be it could be attempted later ?

In case of DHCP, if the IP address changes what is the mechanism to push the new IP to the Master ?